gpu4pyscf/pop/esp.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2024 The GPU4PySCF Authors. All Rights Reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+'''
+references:
+https://onlinelibrary.wiley.com/doi/abs/10.1002/qua.26035
+'''
+
+import numpy as np
+from scipy.spatial import distance_matrix
+import cupy
+from pyscf import gto
+from pyscf.data import radii
+from gpu4pyscf.df import int3c2e
+from gpu4pyscf.lib.cupy_helper import dist_matrix
+
+#modified_Bondi = radii.VDW.copy()
+#modified_Bondi[1] = 1.1/radii.BOHR      # modified version
+
+# Van der Waals radii (in angstrom) are taken from GAMESS.
+R_VDW = 1.0/radii.BOHR * np.asarray([
+    -1,
+    1.20, # H
+    1.20, # He
+    1.37, # Li
+    1.45, # Be
+    1.45, # B
+    1.50, # C
+    1.50, # N,
+    1.40, # O
+    1.35, # F,
+    1.30, # Ne,
+    1.57, # Na,
+    1.36, # Mg
+    1.24, # Al,
+    1.17, # Si,
+    1.80, # P,
+    1.75, # S,
+    1.70]) # Cl
+
+def unit_surface(n):
+    '''
+    Generate spherical harmonics grid points on unit sphere
```

Comment:
1. It's not spherical harmonics grids. Just one kind of "uniform" grids. Is it right? 
Is there a link or a reference for the grids generation scheme.

2. Why not using Lebedev grids?

Reply:
1. Right. The reference is on the top of the file.
2. For reproducibility. The 'uniform' was used qchem, psi4, and GAMESS. Q-Chem also provides the support of Lebedev grids. We can add the support if requested.